src/views/routes/users.js

```diff
@@ -13,12 +14,31 @@ router.post('/register', async (req, res) => {
   }
 });
 
+// eslint-disable-next-line consistent-return
+router.post('/login', async (req, res) => {
+  try {
+    const { email, password } = req.body;
+    const { token, userExists } = await generateToken(email, password);
+
+    if (!userExists) {
+      return responseHandler.handleError(res, 404, 'The user is not registered');
+    }
+    if (!token) {
+      return responseHandler.handleError(res, 401, 'Password incorrect');
```

Comment:
Here you're returning 401 ("password incorrect") if the generateToken() method doesn't return a token. 

You want to handle "password incorrect" more granularly. Currently, if the `generateToken` method doesn't return the token for any reason, this code assumes that it is only because the password is incorrect, which may not be the case. Instead you want to handle the "password incorrect" case inside the `generateToken` method, then throw a specific error that can be caught and interpreted by this /login handler, which returns 401.